support/on-prem-bootstrap.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# The purpose of this script is to download the latest stable version of every
+# package in the core-plans repo, tar them up, and upload to S3. It also supports
+# downloading the archive from S3, extracting it, and uploading to a new depot.
+
+set -e
+
+help() {
+    echo "Usage: on-prem-bootstrap.sh {create-archive|populate-depot <DEPOT_URL>}"
+}
+
+exists() {
+  if command -v $1 >/dev/null 2>&1
+  then
+    return 0
+  else
+    return 1
+  fi
+}
+
+required_tools=( aws git curl jq )
+required_vars=( AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY HAB_AUTH_TOKEN )
+upstream_depot="https://bldr.habitat.sh"
+bucket="habitat-on-prem-builder-bootstrap"
+bootstrap_file="on-prem-bootstrap.tar.gz"
```

Comment:
Might be good to append a timestamp so we can have different versions of the package bundles...

Reply:
Sure, I can do that, though it complicates the download process, as then we need to fetch everything from S3 and iterate through all of the files in order to find the correct one to download.  As the number of files in S3 grows larger, that's going to cause a bigger and bigger delay downloading files.

Reply:
We could do it exactly the how the hab bootstrap bundle script does it, with a LATEST pointer.. copy/pastable code from the other script... a bit more work but would provide some more flexibility.

```
manifest_url=${s3_root_url}/LATEST
log "Downloading latest builder tarball manifest from ${manifest_url}"
${curl} --remote-name ${manifest_url} >&2
latest_archive=$(${awk} 'NR==1' LATEST)
```

Reply:
Ah, nice.  Will do.